plugin/type_inet/sql_type_inet.cc

```diff
@@ -500,7 +501,7 @@ size_t Inet6::to_string(char *dst, size_t dstsize) const
       //
       // If it is not the last field, append closing ':'.
 
-      p += sprintf(p, "%x", ipv6_words[i]);
+      p += snprintf(p, p_end-p, "%x", ipv6_words[i]);
```

Comment:
As in https://github.com/MariaDB/server/pull/3075#discussion_r1496403722, I think `p_end-p` is generally harder for a reader to understand than simply using `dstsize` here.

Reply:
`dstsize` wouldn't be correct since `p` moves around, but I didn't realize there is already a `dstsize_available` variable. Updated this to use that and also assert if the buffer is too short.